lib/tree-view.coffee

```diff
@@ -588,8 +588,15 @@ class TreeView extends View
             if initialPathIsDirectory
               newPath = "#{originalNewPath}#{fileCounter.toString()}"
             else
-              fileArr = originalNewPath.split('.')
-              newPath = "#{fileArr[0]}#{fileCounter.toString()}.#{fileArr[1]}"
+              if process.platform is 'win32'
+                fileArr = /(.+)\.(?!.*\\)(.+)/.exec(originalNewPath)
```

Comment:
I initially tried to use an extended regex pattern here (the `///` provided by Coffeescript), but that led to some nasty escaping issues with `#{path.sep}` because apparently interpolation happens before the regex is parsed.